### PR TITLE
Fix #3243 - Staff naming permission fix

### LINKED ISSUE #3243: Changing staff's name is a GUEST command (permissions)
**OS: Win 7** 
**Version: 0.4.0.0** 
**Commit: a8efcef** 

When you edit a staff members name online, it counts as a **Guest** command, therefor restricting it if the player ain't allowed to edit Guest commands.
The rest of the stuff (hire staff, patrol areas & tell them what to do) is all staff commands
The names should be too, if possible

- [ ] Reproducable in RCT2 (vanilla)?
- [x] Multiplayer?

**Steps to reproduce:**
1. Host a server and join as a client
2. Edit a staffs name
3. See what command the user did ('Guest')

**Screenshots / Video:**
staff

### PR DESCRIPTION
**Pull request by Nubbie**
This fixes issue #3243 

This code was made by another user (Doesn't want to create a Github account)

This has been tested and it works

### CODE REVIEW

**Comment by Nubbie** [comment]
Great one check wasn't successful.. what now

File: src/peep/peep.c [hunk 1]
```diff
@@ -10614,4 +10614,7 @@ void game_command_set_peep_name(int *eax, int *ebx, int *ecx, int *edx, int *esi
 		(uint8*)ebp,
 		(uint8*)edi
 		);
+	}
```
Comment (LinusU) [inline src/peep/peep.c]: This line shouldn't be indented

Comment (Nubbie) [inline src/peep/peep.c]: Are you sure about that? 
Should it be removed completely?

Comment (HaasJona) [inline src/peep/peep.c]: > Without the indent there, the code doesn't work

Indentation should never change if the code works or not. This is C, not python.

Comment (Nubbie) [inline src/peep/peep.c]: 'TAB' removed

**Comment by Nubbie** [comment]
Hope there was nothing else needed to be fixed ^^

File: src/peep/peep.c [hunk 2]
```diff
@@ -10615,3 +10615,6 @@ void game_command_set_peep_name(int *eax, int *ebx, int *ecx, int *edx, int *esi
 		(uint8*)edi
 		);
 }
+ void game_command_set_staff_name(int *eax, int *ebx, int *ecx, int *edx, int *esi, int *edi, int *ebp) {
```
Comment (Gymnasiast) [inline src/peep/peep.c]: You need to remove the space in front of `void`, and put a newline before it.

**Comment by Nubbie** [comment]
Done

**Comment by Gymnasiast** [comment]
Looks good to me. @IntelOrca or @duncanspumpkin could one of you review and merge it?

**Comment by HaasJona** [comment]
This isn't really secure as you can still call `game_command_set_staff_name` on a guest or the other way around (with a hacked client), but apart from that it looks good.

**Comment by IntelOrca** [comment]
To prevent what @HaasJona said can happen, you can add some easy validation so that it checks if the peep is a guest or a staff before performing the action.

**Comment by Nubbie** [comment]
The only problem is that it seems like the user who made this doesn't want to continue working on this code, too busy with other things and Open-Source wasn't his thing.
If this can't be merged right now, anyone is more than welcome to take this as inspiration or fix the last pieces that is needed before getting merged.

**Comment by Gymnasiast** [comment]
I'll pick it up then. Please keep the PR open and leave the branch is it is for now.

**Pull request closed without merging**